lib/engine/game/g_18_zoo.rb

```diff
@@ -0,0 +1,668 @@
+        @companies_for_wednesday = @companies.drop(draw_size + 12).first(4)
+
+        @available_companies.concat(@companies_for_isr)
+
+        if @all_private_visible
+          @log << 'All powers visible in the future deck'
+          @future_companies.concat(@companies_for_monday + @companies_for_tuesday + @companies_for_wednesday)
+        else
+          @future_companies.concat(@companies_for_monday)
+        end
+
+        reserve_shares(false)
+      end
+
+      def init_optional_rules(optional_rules)
+        rules = super
+
+        maps = rules.select { |rule| rule.start_with?('map_') }
+        raise GameError, 'Please select a single map.' unless maps.size <= 1
+
+        @map = maps.empty? ? :map_a : maps.first
+
+        @near_families = @players.size < 5
+        @all_private_visible = rules.include?(:power_visible)
+
+        rules
+      end
+
+      # use to modify hexes based on optional rules
+      def optional_hexes
+        self.class::HEXES_BY_MAP[@map]
+      end
+
+      # use to modify location names based on optional rules
+      def location_name(coord)
+        self.class::LOCATION_NAMES_BY_MAP[@map][coord]
+      end
+
+      def self.title
+        '18ZOO'
+      end
+
+      def unowned_purchasable_companies(_entity)
+        @available_companies + @future_companies
+      end
+
+      private
+
+      def init_round
+        Round::Draft.new(self, [Step::G18ZOO::SimpleDraft], reverse_order: true)
+      end
+
+      def num_trains(train)
+        num_players = @players.size
+
+        case train[:name]
+        when '2S'
+          num_players
+        else
+          super
+        end
+      end
+
+      def init_corporations(stock_market)
+        self.class::CORPORATIONS.select { |corporation| CORPORATIONS_BY_MAP[@map].include?(corporation[:sym]) }
+                                .map do |corporation|
+          Corporation.new(
+            min_price: stock_market.par_prices.map(&:price).min,
+            capitalization: self.class::CAPITALIZATION,
+            coordinates: CORPORATION_COORDINATES_BY_MAP[@map][corporation[:sym]],
+            **corporation.merge(corporation_opts),
+          )
+        end
+      end
+
+      def init_starting_cash(players, bank)
+        hash = SMALL_MAP.include?(@map) ? self.class::STARTING_CASH_SMALL_MAP : self.class::STARTING_CASH_BIG_MAP
+        cash = hash[players.size]
+
+        players.each do |player|
+          bank.spend(cash, player)
+        end
+      end
+
+      def custom_end_game_reached?
+        @turn == 3
+      end
+
+      def new_stock_round
+        result = super
+
+        add_cousins if @turn == 3
+
+        update_current_and_future(@companies_for_monday, @companies_for_tuesday, 1)
+        update_current_and_future(@companies_for_tuesday, @companies_for_wednesday, 2)
+        update_current_and_future(@companies_for_wednesday, nil, 3)
+
+        @available_companies.select { |c| c.owner.nil? }.each { |c| c.owner = @bank }
```

Comment:
@available_companies.each { |c| c.owner = @bank unless c.owner }

Reply:
done in 558751e5